src/main/java/org/example/LessonDao.java

```
@@ -0,0 +1,110 @@
+                String lessonName = resultSet.getString(2);
+                int homeworkId = resultSet.getInt(3);
+                String homeworkName = resultSet.getString(4);
+                String homeworkDescription = resultSet.getString(5);
+
+                Homework homework = new Homework(homeworkId, homeworkName, homeworkDescription);
+                Lesson lesson = new Lesson(lessonId, lessonName, homework);
+
+                lessons.add(lesson);
+            }
+        } catch (SQLException e) {
+            throw new SQLException("Error retrieving lessons: " + e.getMessage());
+        }
+
+        return lessons;
+    }
+
+    public Lesson getLessonById(int lessonId) throws SQLException {
+        String query = "SELECT Lesson.id, Lesson.name, Homework.id, Homework.name, Homework.description " +
+                "FROM Lesson " +
+                "JOIN Homework ON Lesson.homework_id = Homework.id " +
+                "WHERE Lesson.id = ?";
+
+        try (Connection connection = DataBaseConnection.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+
+            statement.setInt(1, lessonId);
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    int lessonIdResult = resultSet.getInt(1);
+                    String lessonName = resultSet.getString(2);
+                    int homeworkId = resultSet.getInt(3);
+                    String homeworkName = resultSet.getString(4);
+                    String homeworkDescription = resultSet.getString(5);
+
+                    Homework homework = new Homework(homeworkId, homeworkName, homeworkDescription);
+                    return new Lesson(lessonIdResult, lessonName, homework);
+                }
+            }
+        } catch (SQLException e) {
+            throw new SQLException("Error retrieving lesson: " + e.getMessage());
+        }
+
+        return null;
```

Instead of returning null you can throw an exception that lesson is not found